[quant] Make APoTObserver inherit from ObserverBase

asl3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79198
* #78905
* __->__ #79068

### Summary:
This PR removes the class "NonUniformQuantizationObserverBase" as the extra inheritance level is not yet needed (following YAGNI https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). It has APoTObserver instead inherit from ObserverBase.

### Test Plan:
python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py

asl3 committed:
[quant] Make APoTObserver inherit from ObserverBase

[ghstack-poisoned]

dzdang commented:
I think you should stack 78905 on top of this one

asl3 committed:
Update on "[quant] Make APoTObserver inherit from ObserverBase"


### Summary:
This PR removes the class "NonUniformQuantizationObserverBase" as the extra inheritance level is not yet needed (following YAGNI https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). It has APoTObserver instead inherit from ObserverBase.

### Test Plan:
python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Make APoTObserver inherit from ObserverBase"


### Summary:
This PR removes the class "NonUniformQuantizationObserverBase" as the extra inheritance level is not yet needed (following YAGNI https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). It has APoTObserver instead inherit from ObserverBase.

### Test Plan:
python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py

[ghstack-poisoned]

dzdang reviewed (approved):
nit: you could add a TODO comment saying that we can consider adding a NonUniformQuantizationObserverBase class when more than 1 non uniform method is implemented

asl3 committed:
Update on "[quant] Make APoTObserver inherit from ObserverBase"


### Summary:
This PR removes the class "NonUniformQuantizationObserverBase" as the extra inheritance level is not yet needed (following YAGNI https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). It has APoTObserver instead inherit from ObserverBase.

### Test Plan:
python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Make APoTObserver inherit from ObserverBase"


### Summary:
This PR removes the class "NonUniformQuantizationObserverBase" as the extra inheritance level is not yet needed (following YAGNI https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). It has APoTObserver instead inherit from ObserverBase.

### Test Plan:
python pytorch/test/quantization/core/experimental/test_nonuniform_observer.py

[ghstack-poisoned]

malfet commented:
Oops, the landed PR title is really weird. @asl3 do you mind if I'll revert it and re-land with the better title?
